optimize/functions/functions.go

```diff
@@ -553,7 +553,7 @@ func (BrownBadlyScaled) Grad(grad, x []float64) {
 
 	f1 := x[0] - 1e6
 	f2 := x[1] - 2e-6
-	f3 := x[0]*x[1] - 2
+	f3 := float64(x[0]*x[1]) - 2
```

Comment:
perhaps add a comment about that? (or just mention `#1218`?)

Reply:
I've put a tag comment that should be enough for people to look at the commit message for this line.